### PR TITLE
Changed supervisor per runit

### PR DESCRIPTION
**Pull request by erseco**
Change the  `supervisor` process manager that is Python3 based to a native `runit` approach that reduces the memory footprint and also the image size (42M to 25M)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,11 +38,15 @@ USER nobody
 WORKDIR /var/www/html
 COPY --chown=nobody src/ /var/www/html/
 
+# Add runit boot scripts
+COPY --chown=nobody config/nginx.run /etc/service/nginx/run
+COPY --chown=nobody config/php.run /etc/service/php/run
+
```
Comment (reishou) [inline Dockerfile:44]: I think we should chmod it
`RUN chmod +x /etc/service/nginx/run \
    && chmod +x /etc/service/php/run`

Comment (erseco) [reply Dockerfile:44]: The files already have the executable bit, but we can add this line to be secure

Comment (reishou) [reply Dockerfile:44]: I don't know why it throw acess denied, when I chmod them, docker run successful

image

Comment (reishou) [reply Dockerfile:44]: Thank you about amazing pull request <3
image

Comment (erseco) [reply Dockerfile:44]: Added

**Comment by TrafeX** [comment]
Hi @erseco,

Thank you for improving this!
Sorry to keep you waiting for so long, the thing is that I'm in doubt of merging this.
It indeed decreases the image size a lot which is very nice. But it increases the complexity and maintainability a bit because it adds a custom script.
I'll have to familiarize myself a bit more with runit to make a decision about this. I'll come back to it!

**Comment by getvivekv** [comment]
Have you considered supervisor go mod ? It's way lesser in size as you don't need those python packages

**Comment by erseco** [comment]
@getvivekv yes, I tested so many process managers, the golang version of `supervisord` is not so stable and the static version is 15Mb in size instead the 284kb of `runit`, also is in the official Alpine repositories, so for sure will be more stable and secure: https://pkgs.alpinelinux.org/package/v3.12/community/x86_64/runit

**Comment by getvivekv** [comment]
@erseco I have tested supervisord gomod for a couple of the projects with no issues. But I agree with you, even though it has been around and in development for a couple of years, it is still not 1.x release yet, so it may not be considered as stable. I am testing runit now, so far looks good.

**Comment by TrafeX** [comment]
Do you also have experience with s6? I like that one as well, the only downside I see is that you need to start s6 as root. It can then drop privileges for the processes it starts, but I would rather drop privileges in the image as soon as possible.

**Comment by erseco** [comment]
Yes, previously to decide to runit I was comparing openrc, upstart, s6, monit and many others that I don't remember and the easiest and lightweight was runit. The official Alpine one is openrc but like s6 has so many dependencies and many configuration files, increasing the image size and adding more complexity. I checked many reviews and comparisons between init systems but I didn't found them, I can try to get that links if you want 😊

**Comment by polarathene** [comment]
`s6-overlay` is only 2MB compressed , or `s6` with it's dependencies is about ~3MB. `runit` is notably lighter weight if shaving off that difference is worthwhile. [This s6 comparison page](https://skarnet.org/software/s6/why.html) doesn't raise any major disadvantages for runit, though it seems like s6 is generally a better choice.

**Comment by TrafeX** [comment]
Hi! I've decided to switch to s6 instead of runit to keep the complexity low and still lower the image size.
Thank you for your help!

**Pull request closed without merging**